jetstream/pull.go

```diff
@@ -388,10 +388,17 @@ func (s *pullSubscription) incrementDeliveredMsgs() {
 // the buffer to trigger a new pull request.
 // lock should be held before calling this method
 func (s *pullSubscription) checkPending() {
-	if s.pending.msgCount < s.consumeOpts.ThresholdMessages ||
-		(s.pending.byteCount < s.consumeOpts.ThresholdBytes && s.consumeOpts.MaxBytes != 0) &&
-			atomic.LoadUint32(&s.fetchInProgress) == 1 {
-		batchSize := s.consumeOpts.MaxMessages - s.pending.msgCount
+	if (s.pending.msgCount < s.consumeOpts.ThresholdMessages ||
+		(s.pending.byteCount < s.consumeOpts.ThresholdBytes && s.consumeOpts.MaxBytes != 0)) &&
+		atomic.LoadUint32(&s.fetchInProgress) == 0 {
+		var batchSize int
+		if s.consumeOpts.MaxBytes == 0 {
+			// if using messages, calculate appropriate batch size
+			batchSize = s.consumeOpts.MaxMessages - s.pending.msgCount
+		} else {
+			// if using bytes, use the max value
```

Comment:
Are we sure we're properly checking for bytes in bytes use case elsewhere?

Reply:
Thanks, there was indeed an issue...